test/persistence/compaction/compaction.td

```diff
@@ -21,147 +21,27 @@ $ set-sql-timeout duration=30s
     WHERE metric = grp.metric
     ORDER BY time DESC LIMIT 1)
 
-# At the beginning there are no trace batches.
-> SELECT value FROM most_recent_mz_metrics where metric = 'mz_persist_trace_blob_count'
-value
------
-0
-
 > CREATE TABLE compaction (f1 INTEGER);
 
-> INSERT INTO compaction VALUES (1);
-
-# The insert should make it into trace, and form 1 trace batch.
-> SELECT value FROM most_recent_mz_metrics where metric = 'mz_persist_trace_blob_count'
-value
------
-1
-
-# Read after write to coerce different writes to happen at different timestamps.
-> SELECT * FROM compaction
-f1
-----
-1
-
-> INSERT INTO compaction VALUES (2);
-
-> SELECT * FROM compaction
-f1
-----
-1
-2
-
-# The previous insert should get compacted into one batch with the one before it.
-> SELECT value FROM most_recent_mz_metrics where metric = 'mz_persist_trace_blob_count'
-value
------
-1
-
-> INSERT INTO compaction VALUES (3);
-
-> SELECT * FROM compaction
-f1
-----
-1
-2
-3
-
-# The most recent insert forms a new batch because it is at a lower compaction level.
-> SELECT value FROM most_recent_mz_metrics where metric = 'mz_persist_trace_blob_count'
-value
------
-2
-
-> INSERT INTO compaction VALUES (4);
-
-> SELECT * FROM compaction
-f1
-----
-1
-2
-3
-4
-
-# The previous insert should compact together with the previous batch, which then compacts recursively
-# up, leaving just one batch.
-> SELECT value FROM most_recent_mz_metrics where metric = 'mz_persist_trace_blob_count'
-value
+# At the beginning there's fewer than 200 bytes in the stored arrangement.
+> SELECT value < 200 AS result FROM most_recent_mz_metrics where metric = 'mz_persist_trace_blob_bytes'
+result
 -----
-1
-
-> INSERT INTO compaction VALUES (5);
+true
 
-> SELECT * FROM compaction
-f1
-----
-1
-2
-3
-4
-5
+> INSERT INTO compaction VALUES (0), (1), (2), (3), (4), (5), (6), (7), (8), (9);
 
-# The previous insert forms a new batch.
-> SELECT value FROM most_recent_mz_metrics where metric = 'mz_persist_trace_blob_count'
-value
+# There's at least 600 bytes in the stored arrangement.
+> SELECT value > 600 AS result FROM most_recent_mz_metrics where metric = 'mz_persist_trace_blob_bytes'
```

Comment:
I'm a little worried about flakiness here because the actual size of the data is small compared to our size overhead. I.e. unless I'm misunderstanding something, I think the vast majority of the difference between 200 and 600 is parquet metadata, the current size of which this test is now accidentally relying on.

Is it possible to construct a single value in SQL that's very large compared to any amount of storage space overhead we'll ever have? I'd imagine something like 1MiB would be a good balance between largeness and test runtime

Reply:
so i tried a couple of iterations of queries like

```
insert into compaction select c2.f1 * 10 + c1 from compaction as c1, compaction as c2
```

where that starts to mess up is -- you have to first seed compaction with some initial data, and now you've done two batches to insert things, and one batch to delete, and everything doesn't compact as expected. I iterated through 4 - 5 different options here (like 3 insertions followed by one deletion) and nothing reliably compacted as expected (because the insertions sometimes have duplicates, which messes with compaction). Let me see if i can insert with generate series or something like that. 

But

Reply:
oh maybe `repeat` would also work here

Reply:
yeah I'm vaguely imagining a single big string value

Reply:
repeat worked great! changed the test to send a ~ 1mb string, and then delete it